File: ios-native/Jottery/Jottery/Views/WebCalcEditorView.swift
import SwiftUI

// MARK: - Calculator Editor (WebView)

/// CodeMirror 6 + math.js REPL calculator editor, loaded from the calc.html bundle.
/// Results are displayed inline next to each expression.
struct WebCalcEditorView: View {
    @Binding var content: String
    var fontSize: CGFloat = EditorTheme.defaultFontSize

    var body: some View {
        WebEditorView(bundleName: "calc", content: $content, fontSize: fontSize)
    }
}

File: android-native/app/src/main/java/com/jottery/android/ui/webview/WebCalcEditorView.kt
package com.jottery.android.ui.webview

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.jottery.android.ui.theme.LocalIsDarkTheme

/**
 * Calculator editor using the calc.html WebView bundle.
 */
@Composable
fun WebCalcEditorView(
    content: String,
    fontSize: Float = 16f,
    onContentChanged: (String) -> Unit,
    modifier: Modifier = Modifier,
) {
    WebEditorView(
        htmlAsset = "calc.html",
        content = content,
        isDark = LocalIsDarkTheme.current,
        fontSize = fontSize,
        onContentChanged = onContentChanged,
        modifier = modifier,
    )
}
